.changeset/slow-roses-call.md

```diff
@@ -0,0 +1,23 @@
+---
+'astro': patch
+---
+
+Adds a new optional `experimental.env.validateSecretsOnStart` property
+
+If enabled, secrets will also be checked on start (dev/build modes). This is useful for example in some CIs to make sure all your secrets are correctly set before deploying.
```

Comment:
Once the docs description is finalized, I'll suggest a shorter version for this line!